db/changes.go

```diff
@@ -347,7 +347,7 @@ func (db *DatabaseCollectionWithUser) wasDocInChannelPriorToRevocation(ctx conte
 	// Iterate over the channel history information on the document and find any periods where the doc was in the
 	// channel and the channel was accessible by the user
 	for _, docHistoryEntry := range append(syncData.ChannelSet, syncData.ChannelSetHistory...) {
-		if docHistoryEntry.Name != chanName {
+		if docHistoryEntry.Name != chanName && "*" != chanName {
```

Comment:
To save evaluating `"*" != chanName` for each iteration it might be worth doing something like this:

```go
isStarChan = chanName == channels.UserStarChannel
for _, ... range ... {
    if !isStarChan && docHistoryEntry.Name != chanName {
```

(the ordering of the statements in the if is also significant, as it gives an early exit opportunity without evaluating `docHistoryEntry.Name != chanName` each loop